activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -641,6 +648,12 @@ def rollback_transaction(transaction = nil)
       end
 
       def within_new_transaction(isolation: nil, joinable: true)
+        old_isolation = current_transaction.isolation
+        if isolation && open_transactions == 1 && current_transaction.fixture_transaction?
```

Comment:
```suggestion
        if isolation && current_transaction.fixture_transaction?
```

Supposedly we no longer need to check the depth.